Add documentation pages for account Linking

jlyon1 opened the pull request.

jlyon1 committed:
Add documentation pages for account Linking

hunterfortuin commented on Writerside/topics/Account-Linking.md:
```diff
@@ -0,0 +1,3 @@
+# Account LInking
+
+Start typing here...
```
I think providing sequence diagrams would be super helpful to help readers wrap their head around what is happening during account linking specifically (but more broadly anywhere we're pushing data around)

hunterfortuin commented on Writerside/topics/Customizing-Account-Login-and-Linking.md:
```diff
@@ -0,0 +1,271 @@
+    const getCatenaSessionID = await fetch(`/api/v1/authentication/session`, {
+      credentials: "include",
+    })
+    const sessionData: { session: string } = await getCatenaSessionID.json()
```
Is this snippet supposed to show storing the session in local storage?

jlyon1 replied:
Updating for clarity

jlyon1 committed:
Capitalization issue.

jlyon1 committed:
Add some documentation for account linking, begin building out doc structure.

- AuthenticationLibrary.topic: Used to store common snippets and terms for the Authentication and auth flows.
- cfg/glossary.xml: Used to store common terms and definitions for the project
- topics/Account-Linking.md: The main topic for account linking, including a high-level overview of the feature and how it works.
- topics/Account-Merging.md: A topic for account merging, which is a sub-feature of account linking.
- topics/Creating-a-New-Account.md: A topic for creating a new account
